src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Circle extends Figure {
+    private double rad;
+    private double area;
+    private String color = new ColorSupplier().getRandomColor();
+
+    public void setFigure() {
```

Comment:
1. Why do you need a method called `setFigure` when you're not setting any figure? You're generating random radius value - you're supposed to pass this value as a constructor parameter.